lib/sles4sap/sap_deployment_automation_framework/configure_workload_tfvars.pm

```diff
@@ -0,0 +1,395 @@
+
+=over
+
+=item * B<job_id>: OpenQA job ID which the deployment belongs to
+
+=item * B<workload_vnet_code>: Workload zone VNET code
+
+=back
+
+=cut
+
+sub define_networking {
+    my (%args) = @_;
+    for my $arg ('job_id', 'workload_vnet_code') {
+        croak "Missing mandatory argument \$args{$arg}" unless $args{$arg};
+    }
+    my %result = (
+        header => '### Networking ###',
+        # The network logical name is mandatory - it is used in the naming convention and should map to the workload virtual network logical name
+        network_logical_name => "$args{workload_vnet_code}",
+        # Workload VNET name - keep as short as possible as resource naming has limitations
+        network_name => "\"OpenQA-$args{job_id}\"",
+        # disable private endpoints for key vaults and storage accounts
+        use_private_endpoint => 'false',
+        # disable service endpoints for key vaults and storage accounts
+        use_service_endpoint => 'true',
+        # Peering between control plane and workload zone (enable connection from deployer VM to SUT network)
+        peer_with_control_plane_vnet => 'true',
+        # Enables firewall for keyvaults and storage - only SUT subnets will be able to access it
+        enable_firewall_for_keyvaults_and_storage => 'true',
+        # Disable resource delete lock for cleanup to work properly
+        place_delete_lock_on_resources => 'false',
+        # Defines if a custom dns solution is used
+        use_custom_dns_a_registration => 'false',
+        # Defines if the Virtual network for the Virtual machines is registered with DNS
+        # This also controls the creation of DNS entries for the load balancers
+        register_virtual_network_to_dns => 'true',
+        # Boolean value indicating if storage accounts and key vaults should be registered to the corresponding dns zones
+        register_storage_accounts_keyvaults_with_dns => 'false',
+        # If defined provides the DNS label for the Virtual Network
+        dns_label => '"openqa.net"'
+    );
+    return \%result;
+}
+
+=head2 define_nat_section
+
+    define_nat_section(environment=>'LAB', sdaf_region=>'swedencentral', workload_vnet_code=>'OpenQA-42');
+
+Returns tfvars section related to NAT setup in B<HASHREF> format.
+B<Example:> {nat_gateway_name : 'NAT-01', network_name : 'OpenQA-NAT01' ... }
+
+=over
+
+=item * B<environment>: SDAF environment
+
+=item * B<sdaf_region>: Public cloud location
+
+=item * B<workload_vnet_code>: Public cloud location
+
+=back
+
+=cut
+
+sub define_nat_section {
+    my (%args) = @_;
+    for my $arg ('environment', 'sdaf_region', 'workload_vnet_code') {
+        croak "Missing mandatory argument \$args{$arg}" unless $args{$arg};
+    }
+    my %result = (
+        header => '###  NAT Configuration ###',
+        deploy_nat_gateway => 'true',
+        nat_gateway_name => "\"$args{environment}-$args{sdaf_region}-$args{workload_vnet_code}-NG_0001\""
```

Comment:
I realize there are too many lines like this to change, but for next time, you can do:
```
        nat_gateway_name => qq|"$args{environment}-$args{sdaf_region}-$args{workload_vnet_code}-NG_0001"|,
```
I think it's easier to read.

Reply:
Agree and done. Will run all 4 VRs at the end of the day just to be sure.